hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,303 @@
+                nodes = hou.nodeType(node_category, node_type).instances()
+
+                for node in nodes:
+
+                    # get the evaluated path parm value
+                    path = node.parm(path_parm_name).eval()
+
+                    # ensure the output path exists
+                    if not os.path.exists(path):
+                        continue
+
+                    self.logger.info(
+                        "Processing %s node: %s" % (node_type, node.path()))
+
+                    # allow the base class to collect and create the item. it
+                    # should know how to handle the output path
+                    item = super(HoudiniSessionCollector, self)._collect_file(
+                        parent_item,
+                        path,
+                        frame_sequence=True
+                    )
+
+                    # the item has been created. update the display name to
+                    # include the node path to make it clear to the user how it
+                    # was collected within the current session.
+                    item.name = "%s (%s)" % (item.name, node.path())
+
+    def collect_tk_alembicnodes(self, parent_item):
+
+        publisher = self.parent
+        engine = publisher.engine
+
+        alembicnode_app = engine.apps.get("tk-houdini-alembicnode")
+        if not alembicnode_app:
+            self.logger.debug(
+                "The tk-houdini-alembicnode app is not installed. "
+                "Will not attempt to collect those nodes."
+            )
+            return
+
+        try:
+            tk_alembic_nodes = alembicnode_app.get_nodes()
+        except AttributeError, e:
+            self.logger.warning(
+                "Unable to query the session for tk-houdini-alembicnode "
+                "instances. It looks like perhaps an older version of the "
+                "app is in use which does not support querying the nodes. "
+                "Consider updating the app to allow publishing their outputs."
+            )
+            return
+
+        for node in tk_alembic_nodes:
+
+            out_path = alembicnode_app.get_output_path(node)
+
+            if not os.path.exists(out_path):
+                continue
+
+            self.logger.info(
+                "Processing sgtk_alembic node: %s" % (node.path(),))
+
+            # allow the base class to collect and create the item. it
+            # should know how to handle the output path
+            item = super(HoudiniSessionCollector, self)._collect_file(
+                parent_item, out_path)
+
+            # the item has been created. update the display name to
+            # include the node path to make it clear to the user how it
+            # was collected within the current session.
+            item.name = "%s (%s)" % (item.name, node.path())
+
+            self._alembic_nodes_collected = True
+
+    def collect_tk_mantranodes(self, parent_item):
```

Comment:
New collection method for tk mantra nodes. Relies on changes to the app to make this simpler.